src/modules/api-client/utils.ts

```diff
@@ -0,0 +1,257 @@
+import { ServerText, TranslatedString, translation } from "@atb/translations";
+import { NextApiRequest, NextApiResponse } from "next";
+import bunyan from "bunyan";
+import { v4 as uuidv4 } from "uuid";
+
+export const logger = bunyan.createLogger({
+  name: "planner-web",
+  streams: [{ stream: process.stdout, level: "info" }],
+  serializers: {
+    req: (req: Request) => {
+      return {
+        method: req.method,
+        url: req.url,
+        headers: {
+          ...req.headers,
+          cookie: null,
+        },
+      };
+    },
+    res: bunyan.stdSerializers.res,
+    upstream: (up: Response) => {
+      return {
+        statusCode: up.status,
+        url: up.url,
+        statusText: up.statusText,
+      };
+    },
+    err: bunyan.stdSerializers.err,
+  },
+});
+
+export function logError(e: any) {
+  if (process.env.NODE_ENV === "development") {
+    console.error(e);
+  } else {
+    logger.error(e);
+  }
+}
+
+export function logApplicationError(
+  e: ApplicationError,
+  req: Request | NextApiRequest,
+  res: Response | NextApiResponse<any>
+) {
+  if (process.env.NODE_ENV === "development") {
+    console.error(e);
+  } else {
+    logger.error(
+      {
+        upstream: e.upstreamResponse,
+        res,
+        req,
+        err: e,
+      },
+      e.message
+    );
+  }
+}
+
+const externalHttpUrls = {
+  entur: "https://api.entur.io",
+} as const;
+
+export type HttpEndpoints = keyof typeof externalHttpUrls;
+
+export type Requester<T extends HttpEndpoints> = (
+  url: `/${string}`,
+  init?: RequestInit | undefined
+) => Promise<Response>;
+
+export function createRequester<T extends HttpEndpoints>(
+  baseUrlKey: T,
+  correlationId: string | undefined
+): Requester<T> {
+  return async function request(
+    url: `/${string}`,
+    init?: RequestInit | undefined
+  ) {
+    const baseUrl = externalHttpUrls[baseUrlKey];
+    const actualUrl = `${baseUrl}${url}`;
+
+    try {
+      const data = await fetch(actualUrl, {
+        ...init,
+        headers: {
+          ...init?.headers,
+          "ET-Client-Name": "FOO",
```

Comment:
Should the client name be updated at a later point?

Reply:
Yep. Just an example. need to plug in env vars etc. Doesn't require a lot though, the point is just that we need a configurable et client. Might construct it from selected org though 🤔

Reply:
That would work. E.g. `<orgId>-planner-web`.